File: MultitenancyDatabase/src/main/java/br/com/devcave/multitenancy/tenant/TenantFilter.java
package br.com.devcave.multitenancy.tenant;

import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Configuration;
import org.springframework.util.StringUtils;

import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.http.HttpServletRequest;
import java.io.IOException;

@Slf4j
@Configuration
public class TenantFilter implements Filter {

    @Override
    public void doFilter(ServletRequest servletRequest,
                         ServletResponse servletResponse,
                         FilterChain filterChain) throws IOException, ServletException {
        log.info("filter");

        String tenantId = ((HttpServletRequest) servletRequest).getHeader("tenantId");
        if (StringUtils.isEmpty(tenantId)){
            tenantId = "postgres";
        }
        Tenant.setIdentificador(tenantId);

        filterChain.doFilter(servletRequest, servletResponse);
    }
}

File: MultitenancyDiscriminator/src/main/kotlin/br/com/devcave/multitenancy/tenant/TenantFilter.kt
package br.com.devcave.multitenancy.tenant

import org.springframework.stereotype.Component
import org.springframework.web.filter.OncePerRequestFilter
import javax.servlet.FilterChain
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Component
class TenantFilter : OncePerRequestFilter() {
    override fun doFilterInternal(
        request: HttpServletRequest,
        response: HttpServletResponse,
        filterChain: FilterChain
    ) {
        logger.info("do filter")
        kotlin.runCatching {
            request.getHeader("tenantId").toLongOrNull()?.let {
                setTenant(it)
            }
        }
        filterChain.doFilter(request, response)
    }
}